File: app/src/main/java/com/meghdut/upsilent/adapters/MainFragmentPager.java
package com.meghdut.upsilent.adapters;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.meghdut.upsilent.fragments.MoviesFragment;
import com.meghdut.upsilent.fragments.TVShowsFragment;

/**
 * Created by Meghdut Mandal on 06/02/17.
 */

public class MainFragmentPager extends FragmentPagerAdapter {
    public MainFragmentPager(FragmentManager fm) {
        super(fm);
    }

    @Override
    public int getCount() {
        return 2;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return "Movies";
            case 1:
                return "TV Shows";
        }
        return null;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new MoviesFragment();

            case 1:
                return new TVShowsFragment();
        }
        return null;
    }

}

File: app/src/main/java/com/meghdut/upsilent/adapters/MainFragmentPager.kt
package com.meghdut.upsilent.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.meghdut.upsilent.fragments.explore.MoviesFragment
import com.meghdut.upsilent.fragments.explore.TVShowsFragment

/**
 * Created by Meghdut Mandal on 06/02/17.
 */
class MainFragmentPager(fm: FragmentManager?) : FragmentPagerAdapter(fm!!) {
    override fun getCount(): Int {
        return 2
    }

    override fun getPageTitle(position: Int): CharSequence? {
        when (position) {
            0 -> return "Movies"
            1 -> return "TV Shows"
        }
        return null
    }

    override fun getItem(position: Int): Fragment {
        when (position) {
            0 -> return MoviesFragment()
            1 -> return TVShowsFragment()
        }
        throw  IllegalArgumentException("The number cant exceed 1")
    }
}